src/services/email_service.ts

```diff
@@ -0,0 +1,70 @@
+import * as Postmark from 'postmark';
+import { renameKeys } from '../utils/ramdaHelpers';
+import { pipe, curry, evolve } from 'ramda';
+
+const templateJson = require('./templates.json');
+
+type Email = {
+  from: string,
+  to: string,
+  templateId: string,
+  templateModel?: object,
+  attachements?: [{
+    name: string,
+    content: string,
+    contentType: string,
+  }],
+};
+
+type RemoteTemplate = {
```

Comment:
What is this for? Is it an attempt to decouple application code from the specific Postmark template numbers?

If so, good idea, but using an enum would be clearer in the calling code:
```ts
enum VisitorEmailTemplate = {
  WELCOME_CB,
  WELCOME_VISITOR,
  NEW_VISITOR_FOR_CB,
  NEW_VISITOR_FOR_TWINE,
  PASSWORD_RESET,
}

enum AdminEmailTemplate = {
  ...
}

type EmailTemplate = VisitorEmailTemplate | AdminEmailTemplate

// Calling code:
import { VisitorEmailTemplate as Template } from '../services/email'

// ...
sendEmail({
  from: 'foo@bar.com',
  to: 'lol@cat.com',
  template: Template.WELCOME_CB,
  ...
})
```

Reply:
Can you make enums relate to data? I'm storing the remote template id's in `templates.json` and grabbing this info from there.  The type alias was just for functions using this data to let them know what they're expecting.

Reply:
I think enums can take integer values (though I can't remember the syntax). You can still do the mapping to template IDs via an object or switch statement, just not through a JSON file.

Reply:
Oooo the internet suggests using the Map object...